File: Groove.SP/groove-sp-web/src/app/core/models/survey/survey-answer.model.ts
import { SurveyQuestionType } from "../enums/enums";
import { SurveyQuestionModel } from "./survey-question.model";

export interface SurveyAnswerModel {
    id: number;
    answerText: string | null;
    answerNumeric: number | null;
    username: string;
    questionId: number;
    surveyId: number;
    optionId: number | null;
    questionType: SurveyQuestionType;
    isIncludeOpenEndedText: boolean;
}

File: Groove.SP/Groove.SP.Core/Entities/SurveyAnswerModel.cs
namespace Groove.SP.Core.Entities
{
    public class SurveyAnswerModel : Entity
    {
        public long Id { get; set; }
        public string AnswerText { get; set; }
        public int? AnswerNumeric { get; set; }
        public string Username { get; set; }
        public long QuestionId { get; set; }
        public long? OptionId { get; set; }

        public SurveyQuestionModel Question { get; set; }
        public SurveyQuestionOptionModel Option { get; set; }
    }
}